### PR TITLE
fix: handle search rectangles crossing the antimeridian

### PR DESCRIPTION
**Pull request by njelich**
Split wrapping longitude ranges into two envelopes so searches near Fiji and other antimeridian locations include both sides without matching the whole world.

Partial fix for search bounding box bug (antimeridian case) #9604 

## Testing
I added a test in `test_search.py` for the edge case, and ran the test suite.

**Backend checklist**
- [x] Added tests for any new code or added a regression test if fixing a bug
- [x] Run the backend locally and it works

## For maintainers
- [x] Maintainers can push commits to my branch
- [x] Maintainers can merge this PR for me

### CODE REVIEW

File: app/backend/src/couchers/servicers/search.py [hunk 1]
```diff
@@ -69,6 +69,33 @@
...
 
 
+def _within_rectangle(geom: Any, rect: search_pb2.RectArea) -> Any:
+    """
+    Builds a geometry filter for a lat/lng rectangle, splitting it in two when it wraps over the antimeridian
+    (lng_min > lng_max), since ST_MakeEnvelope can't represent a wrapping box.
+    """
+
+    def within(lng_min: float, lng_max: float) -> Any:
+        return func.ST_Within(
```
Comment (Gwened) [inline app/backend/src/couchers/servicers/search.py:79]: Couldn't we replace ST_Within by ST_Intersects, and then omit the and_ clause below?

Comment (njelich) [reply app/backend/src/couchers/servicers/search.py]: Replacing ST_Within by ST_Intersects would break the condition in `test_search.py` for a user on the boundary.

```python (test_search.py:L139-L140)
    # outside (not fully inside)
    user5, token5 = generate_user(geom=create_coordinate(0, 0), geom_radius=100)
```

If we are fine with that "regression", then I can replace it, otherwise I would keep it.
